pkg/cmd/pr/edit/edit.go

```diff
@@ -0,0 +1,259 @@
+package edit
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	shared "github.com/cli/cli/pkg/cmd/pr/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+)
+
+type EditOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Remotes    func() (context.Remotes, error)
+	Branch     func() (string, error)
+
+	Surveyor        Surveyor
+	Fetcher         EditableOptionsFetcher
+	EditorRetriever EditorRetriever
+
+	SelectorArg string
+	Interactive bool
+
+	shared.Editable
+}
+
+func NewCmdEdit(f *cmdutil.Factory, runF func(*EditOptions) error) *cobra.Command {
+	opts := &EditOptions{
+		IO:              f.IOStreams,
+		HttpClient:      f.HttpClient,
+		Remotes:         f.Remotes,
+		Branch:          f.Branch,
+		Surveyor:        surveyor{},
+		Fetcher:         fetcher{},
+		EditorRetriever: editorRetriever{config: f.Config},
+	}
+
+	cmd := &cobra.Command{
+		Use:   "edit {<number> | <url>}",
+		Short: "Edit a pull request",
+		Example: heredoc.Doc(`
+			$ gh pr edit 23 --title "I found a bug" --body "Nothing works"
+			$ gh pr edit 23 --label "bug,help wanted"
+			$ gh pr edit 23 --label bug --label "help wanted"
+			$ gh pr edit 23 --reviewer monalisa,hubot  --reviewer myorg/team-name
+			$ gh pr edit 23 --assignee monalisa,hubot
+			$ gh pr edit 23 --assignee @me
+			$ gh pr edit 23 --project "Roadmap"
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.SelectorArg = args[0]
+
+			flags := cmd.Flags()
+			if flags.Changed("title") {
+				opts.Editable.TitleEdited = true
+			}
+			if flags.Changed("body") {
+				opts.Editable.BodyEdited = true
+			}
+			if flags.Changed("reviewer") {
+				opts.Editable.ReviewersEdited = true
+			}
+			if flags.Changed("assignee") {
+				opts.Editable.AssigneesEdited = true
+			}
+			if flags.Changed("label") {
+				opts.Editable.LabelsEdited = true
+			}
+			if flags.Changed("project") {
+				opts.Editable.ProjectsEdited = true
+			}
+			if flags.Changed("milestone") {
+				opts.Editable.MilestoneEdited = true
+			}
+
+			if !opts.Editable.Dirty() {
+				opts.Interactive = true
+			}
+
+			if opts.Interactive && !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("--tile, --body, --reviewer, --assignee, --label, --project, or --milestone required when not running interactively")}
```

Comment:
Instead of listing all flags, could we shorten this message to something like `pass flags to edit fields when not running interactively`?

Reply:
Addressed in https://github.com/cli/cli/pull/2949